httpserver/service_node.h

```diff
@@ -230,10 +238,22 @@ class ServiceNode {
         std::vector<service_node::storage::Item>& all_entries) const;
 
     // Return the current PoW difficulty
-    int get_pow_difficulty() const;
+    pow_difficulty_t get_pow_difficulty() const;
 
     bool retrieve(const std::string& pubKey, const std::string& last_hash,
                   std::vector<service_node::storage::Item>& items);
+
+    void
+    set_difficulty_history(const std::vector<pow_difficulty_t>& new_history) {
+        curr_pow_difficulty_ =
+            pow_difficulty_t{std::chrono::milliseconds{0}, 1};
+        pow_history_ = new_history;
```

Comment:
Maybe this also needs a default hardcoded value in case `set_difficulty_history` never gets called (unless we don't accept any requests until we hear from the dns server)?